lib/ui/window/platform_isolate.cc

```diff
@@ -0,0 +1,105 @@
+// Copyright 2023 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <iostream>
+#include <thread>
+
+#include "../../../../../../../dart-sdk/sdk/runtime/vm/os_thread.h"
+#include "flutter/fml/macros.h"
+#include "flutter/fml/task_runner.h"
+#include "flutter/lib/ui/window/platform_isolate.h"
+#include "flutter/runtime/dart_isolate.h"
+
+namespace flutter {
+
+void PlatformIsolateNativeApi::Spawn(Dart_Handle entry_point,
+                                     Dart_Handle debug_name) {
+  const char* debug_name_cstr;
+  Dart_StringToCString(debug_name, &debug_name_cstr);
+  std::string debug_name_str = debug_name_cstr;
+  const char* eps;
+  Dart_StringToCString(Dart_ToString(entry_point), &eps);
+  std::cout << "PlatformIsolateNativeApi::Spawn\t" << eps << std::endl;
+
+  Dart_Isolate parent_isolate = Dart_CurrentIsolate();
+  ASSERT(parent_isolate != NULL);
+  std::shared_ptr<DartIsolate>* parent_isolate_data =
+      static_cast<std::shared_ptr<DartIsolate>*>(
+          Dart_IsolateData(parent_isolate));
+
+  Dart_ExitIsolate();  // Exit parent_isolate.
+
+  char* error = NULL;
+  // TODO: shutdown_callback and cleanup_callback.
+  Dart_Isolate new_isolate =
+      Dart_CreateIsolateInGroup(parent_isolate, debug_name_str.c_str(),
+                                NULL,  // shutdown_callback,
+                                NULL,  // cleanup_callback,
+                                NULL,  // child_isolate_data,
+                                &error);
+  std::cout << "Dart_CreateIsolateInGroup " << (void*)new_isolate << "\t"
+            << (error ? error : "null") << std::endl;
+  if (error != NULL) {
+    // TODO: Handle error.
+    std::cout << "Failed to create isolate. " << error << std::endl;
+    return;
+  }
+
+  void* child_isolate_data;
+  bool init_ok =
+      DartIsolate::InitializePlatformIsolate(&child_isolate_data, &error);
+  // TODO: Do we need to set this isolate data on the Isolate somehow, or does
+  // InitializeIsolate do this? Might need a new API function for this.
```

Comment:
I think you would need something in the message callback epilogue to notice the isolate is done. I think this doesn't happen for the UI isolates because the UI isolates have event sources other than receive ports.

```c++
MessageNotifyCallback() {
  pending_messages++;
  platform_task_runner.PostTask(OnMessage);
}

OnMessage() {
  ...
  pending_messages--;
  Dart_HandleMessage();
  if (pending_messages == 0 && !Dart_HasLivePorts()) {
    Dart_ShutdownIsolate();
  }
}
```